pilot/pkg/config/memory/store.go

```diff
@@ -206,44 +209,20 @@ func (cr *store) Update(cfg config.Config) (string, error) {
 	if hasConflict(existing.(config.Config), cfg) {
 		return "", errConflict
 	}
+	if cfg.Annotations != nil {
+		cfg.ResourceVersion = cfg.Annotations[ResourceVersion]
+		delete(cfg.Annotations, ResourceVersion)
+	}
+	if cfg.ResourceVersion == "" {
+		cfg.ResourceVersion = time.Now().String()
```

Comment:
don't we need to update resource version?

Reply:
That is how k8s works - when you apply an update the resource version changes.

Reply:
And we can skip it if we hit L212 (annotation with resource version)

Reply:
> That is how k8s works - when you apply an update the resource version changes.

Yes, resource version should change when updating

Reply:
> And we can skip it if we hit L212 (annotation with resource version)

Yes, using if...else may be logically clearer.
L216  The reason for judging whether  resource version is empty is because I am worried that annotations are not empty, but annotations[ResourceVersion] is empty.